src/FakeItEasy/ExceptionThrowerConfigurationExtensions.cs

```diff
@@ -130,6 +134,11 @@ public static IAfterCallSpecifiedConfiguration Throws<T1>(this IExceptionThrower
                 });
         }
 
+#if DISABLED
+
+        // This method already has a generic parameter, and it can't be inferred; if we add another, we make it unusable,
+        // especially since the new one is something unintelligible like IAnyCallConfigurationWithReturnTypeSpecificied<Foo>...
+

```

Comment:
> In fact, migrating all the extension methods to the interface they extend would solve all the usability problems,

Oh, good. That's what I originally thought. I thought you'd seen something that prevented it.

>  Interfaces should be as small as possible; the current approach of having only general purpose methods in the interface and extension methods for specific use cases is clean and elegant.

That's never really been my opinion. I'm all for focused interfaces, but using extension methods to provide core functionality has always felt a little off to me. There's the advantage that you don't have to provide implementations on as many concrete classes, but it seems a little "polyfill" to me. Especially in FIE, where we don't expect users to implement these interfaces. To contrast with the `IEnumerable` extension methods, where everyone's encouraged to provide their own minimal implementation of an interface and the LINQ methods extend it to a rich ecosystem.

If the interfaces were abstract classes would you feel better about it?